src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -432,6 +437,41 @@ function isCustomComponent(tagName, props) {
   return tagName.indexOf('-') >= 0 || props.is != null;
 }
 
+/**
+ * Returns the children of this dom node in a way that is easily consumable for
+ * comparing to the component hierarchy.
+ * Unfortunately, a node's DOM children don't correspond exactly to its component's
+ * children. For example, text nodes in the component tree become 2 commment DOM
+ * nodes and an optional text dom node. Empty components become a single comment DOM node.
+ *
+ * Returns an array which contains domNodes and arrays of domNodes. Each item in
+ * the returned array represents either one or multiple dom nodes that should correspond
+ * to a single child node in the component tree.
+ */
+function getChildrenFromDomNode(parent) {
```

Comment:
Maybe `getRenderedComponentsFromDomNode()`?  When I see a `getChildren` function, I assume the return value is going to match `node.children`.  But this structure is more specific to the way React renders children.  I don't feel strongly.